specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/preview/2021-10-01-preview/appconfiguration.json

```diff
@@ -0,0 +1,2275 @@
+          "description": "The URI that can be used to request the next set of paged results.",
+          "type": "string"
+        }
+      }
+    },
+    "ApiKey": {
+      "description": "An API key used for authenticating with a configuration store endpoint.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "The key ID.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "A name for the key describing its usage.",
+          "type": "string",
+          "readOnly": true
+        },
+        "value": {
+          "description": "The value of the key that is used for authentication purposes.",
+          "type": "string",
+          "readOnly": true
+        },
+        "connectionString": {
+          "description": "A connection string that can be used by supporting clients for authentication.",
+          "type": "string",
+          "readOnly": true
+        },
+        "lastModified": {
+          "format": "date-time",
+          "description": "The last time any of the key's properties were modified.",
+          "type": "string",
+          "readOnly": true
+        },
+        "readOnly": {
+          "description": "Whether this key can only be used for read operations.",
+          "type": "boolean",
+          "readOnly": true
+        }
+      }
+    },
+    "RegenerateKeyParameters": {
+      "description": "The parameters used to regenerate an API key.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "The id of the key to regenerate.",
+          "type": "string"
+        }
+      }
+    },
+    "KeyValueListResult": {
+      "description": "The result of a request to list key-values.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The collection value.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/KeyValue"
+          }
+        },
+        "nextLink": {
+          "description": "The URI that can be used to request the next set of paged results.",
+          "type": "string"
+        }
+      }
+    },
+    "KeyValue": {
+      "description": "The key-value resource along with all resource properties.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "The resource ID.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "The name of the resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "The type of the resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "properties": {
+          "$ref": "#/definitions/KeyValueProperties",
+          "description": "All key-value properties.",
+          "x-ms-client-flatten": true
+        }
+      },
+      "x-ms-azure-resource": true
```

Comment:
> "x-ms-azure-resource": true [](http://example.com/codeflow?start=5&length=28)

All ARM resources should support systemData.
[https://armwiki.azurewebsites.net/api_contracts/ResourceSystemData.html](https://armwiki.azurewebsites.net/api_contracts/ResourceSystemData.html)

Please consider referencing the "Resource" definition in /specification/common-types/resource-management/v3/types.json.

Reply:
Thanks @j5lim.

This resource is a data plane resource surfaced through our management plane to enable ARM template scenarios. Since it lives it originates from our data plane, no system data has been, or is planned to be, tracked for it. We do not plan on adding it. Is it possible to note the planned omission?

Reply:
Approving for now since the api version is preview and the resource is modifiable via data plane as well.